fix doc about NPD description (#740)
kubernetes-sigs/descheduler: docs/user-guide.md

Descheduler's `RemovePodsViolatingNodeTaints` strategy can be combined with

> [Node Problem Detector](https://github.com/kubernetes/node-problem-detector/) and

Reviewer: It seems like NPD doesn't really do anything to help with autohealing, so we could probably just remove it entirely from this section. Or, add a note in here that it could eventually be combined here when https://github.com/kubernetes/node-problem-detector/pull/565 is available in NPD

Author: thanks, i add a note now :)

[Cluster Autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler) to automatically remove

> Nodes which have problems. Node Problem Detector can detect specific Node problems and reports them to apiserver.

Reviewer: nit: `and report them to the API server`

Author: fix done~ please check :)

> There is a feature call TaintNodeByCondition of the node controller that takes some conditions and turns them in to taints and this only works for the default node conditions: PIDPressure, MemoryPressure, DiskPressure, Ready, and some cloud provider specific conditions.

Reviewer: ```suggestion
There is a feature called TaintNodeByCondition of the node controller that takes some conditions and turns them into taints. Currently, this only works for the default node conditions: PIDPressure, MemoryPressure, DiskPressure, Ready, and some cloud provider specific conditions.
```

Author: fix done~ please check :)

> problems. The Descheduler will then deschedule workloads from those Nodes. Finally, if the descheduled Node's resource

Reviewer: ```suggestion
The Descheduler will then deschedule workloads from those Nodes. Finally, if the descheduled Node's resource
```

Author: fix done~ please check :)

Author: may i click resolve conversation?
